### PR TITLE
README overhaul

### PR DESCRIPTION
**Pull request by chrisarcand**
## Description

Revamps the README a bit, bringing some things up-to-date, reorganizing everything a bit more sensibly, and relentlessly cutting out some quickly-out-of-date details that are canonically written elsewhere and directing people to those sources.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,40 +78,29 @@ To use this provider binary, you have a few different options:
...
-    tfe = "~> 0.15.0"
-  }
-}
-```
+[Create a user or team API token in Terraform Cloud/Enterprise](https://www.terraform.io/docs/cloud/users-teams-organizations/api-tokens.html), and use the token in the provider configuration block:
 
-If you are using Terraform CLI version 0.11.x, you can constrain this provider to 0.15.x versions 
-by adding the version constraint to the tfe provider block.
-```
+```hcl
```
Comment (radditude) [inline README.md:86]: TIL that Github supports syntax highlighting for HCL!

**Review by radditude — APPROVED** [review]
This is such an improvement!

File: README.md (hunk 1 above)
Comment (chrisarcand) [reply README.md:86]: Totally guessed, I honestly have no idea lol

Comment (chrisarcand) [reply README.md:86]: Looks like it! https://github.com/github/linguist/blob/65d078bab8a8bafc5c53614be5c3fec42c2897f0/lib/linguist/languages.yml#L2016-L2030

**Pull request merged**